javascript/ql/lib/semmle/javascript/frameworks/NoSQL.qll

```diff
@@ -17,490 +17,73 @@ module NoSql {
-        or
-        name = "find" and n = 0
-        or
-        name = "findOne" and n = 0
-        or
-        name = "findOneAndDelete" and n = 0
-        or
-        name = "findOneAndRemove" and n = 0
-        or
-        name = "findOneAndReplace" and n = 0
-        or
-        name = "findOneAndUpdate" and n = 0
-        or
-        name = "remove" and n = 0
-        or
-        name = "replaceOne" and n = 0
-        or
-        name = "update" and n = 0
-        or
-        name = "updateMany" and n = 0
-        or
-        name = "updateOne" and n = 0
-      )
+  /** A call to a MongoDB query method. */
+  private class QueryCall extends DatabaseAccess, API::CallNode {
+    QueryCall() {
+      this = ModelOutput::getATypeNode("mongodb", "Collection").getAMember().getACall() and
+      not this.getCalleeName() = ["toString", "valueOf", "getLogger"]
       or
-      // UpdateQuery
-      (
-        name = "findOneAndUpdate" and n = 1
-        or
-        name = "update" and n = 1
-        or
-        name = "updateMany" and n = 1
-        or
-        name = "updateOne" and n = 1
-      )
-    }
-  }
-}
-
-/**
- * Provides classes modeling the Mongoose library.
- */
-private module Mongoose {
-  /**
-   * Gets an import of Mongoose.
-   */
-  API::Node getAMongooseInstance() { result = API::moduleImport("mongoose") }
-
-  /**
-   * Gets a reference to `mongoose.createConnection`.
-   */
-  API::Node createConnection() { result = getAMongooseInstance().getMember("createConnection") }
-
-  /**
-   * A Mongoose function.
-   */
-  abstract private class MongooseFunction extends API::Node {
-    /**
-     * Gets the API-graph node for the result from this function (if the function returns a `Query`).
-     */
-    abstract API::Node getQueryReturn();
-
-    /**
-     * Holds if this function returns a `Query` that evaluates to one or
-     * more Documents (`asArray` is false if it evaluates to a single
-     * Document).
-     */
-    abstract predicate returnsDocumentQuery(boolean asArray);
-
-    /**
-     * Gets an argument that this function interprets as a query.
-     */
-    abstract API::Node getQueryArgument();
-  }
-
-  /**
-   * Provides classes modeling the Mongoose Model class
-   */
-  module Model {
-    private class ModelFunction extends MongooseFunction {
-      string methodName;
-
-      ModelFunction() { this = getModelObject().getMember(methodName) }
-
-      override API::Node getQueryReturn() {
-        MethodSignatures::returnsQuery(methodName) and result = this.getReturn()
-      }
-
-      override predicate returnsDocumentQuery(boolean asArray) {
-        MethodSignatures::returnsDocumentQuery(methodName, asArray)
-      }
-
-      override API::Node getQueryArgument() {
-        exists(int n |
-          MethodSignatures::interpretsArgumentAsQuery(methodName, n) and
-          result = this.getParameter(n)
-        )
-      }
+      this =
+        ModelOutput::getATypeNode("mongodb", ["Db", "MongoClient"])
```

Comment:
```suggestion
        ModelOutput::getATypeNode("mongodb", "Db")
```

In the above you alias MongoClient as DB (the `mongodb;Db;mongodb;MongoClient;` row).  
So this suggestion should be equivalent? 

(Maybe you did it like you've done on purpose, and in that case dismiss this suggestion).

Reply:
Yes, this was on purpose. The 2.x adapter was meant to stand on its own so the rest of the model wouldn't need to worry about it too much. If we later change how the 2.x adapter works, it would be a hassle to have to fix up places where we assumed they would be aliased.